core/unit_test/TestTeamBasic.hpp

```diff
@@ -396,6 +394,92 @@ TEST(TEST_CATEGORY, team_broadcast_double) {
   }
 }
 
+struct TeamBroadcastIntPtrFunctor {
+  using TeamMember = typename Kokkos::TeamPolicy<TEST_EXECSPACE>::member_type;
+
+  KOKKOS_FUNCTION void operator()(const TeamMember& team_member) const {
+    int* ptr = team_member.team_rank() == 0 ? view.data() : nullptr;
+    team_member.team_broadcast(ptr, 0);
+    if (ptr != view.data()) errors()++;
+  }
+
+  Kokkos::View<int, TEST_EXECSPACE> view;
+  Kokkos::View<int, TEST_EXECSPACE, Kokkos::MemoryTraits<Kokkos::Atomic>>
+      errors;
+};
+
+TEST(TEST_CATEGORY, team_broadcast_int_ptr) {
+  Kokkos::View<int, TEST_EXECSPACE> view("view");
+  Kokkos::View<int, TEST_EXECSPACE, Kokkos::MemoryTraits<Kokkos::Atomic>>
+      errors("errors");
+  TeamBroadcastIntPtrFunctor functor{view, errors};
+  auto team_size = Kokkos::TeamPolicy<TEST_EXECSPACE>(1, Kokkos::AUTO)
+                       .team_size_max(functor, Kokkos::ParallelForTag());
+  Kokkos::parallel_for(Kokkos::TeamPolicy<TEST_EXECSPACE>(1, team_size),
+                       functor);
+  Kokkos::fence();
+  ASSERT_EQ(Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace{}, errors)(),
+            0);
```

Comment:
I'd prefer if you moved that logic into `TeamBroadcastIntPtrFunctor` and if you used a reduction instead of an atomic view that you mirror.